indexing-service/src/main/java/io/druid/indexing/firehose/IngestSegmentFirehoseFactory.java

```diff
@@ -143,14 +145,8 @@ public Firehose connect(InputRowParser inputRowParser, File temporaryDirectory)
           .getTaskActionClient()
           .submit(new SegmentListUsedAction(dataSource, interval, null));
       final Map<DataSegment, File> segmentFileMap = taskToolbox.fetchSegments(usedSegments);
-      VersionedIntervalTimeline<String, DataSegment> timeline = new VersionedIntervalTimeline<>(
-          Comparators.naturalNullsFirst()
-      );
-
-      for (DataSegment segment : usedSegments) {
-        timeline.add(segment.getInterval(), segment.getVersion(), segment.getShardSpec().createChunk(segment));
-      }
-      final List<TimelineObjectHolder<String, DataSegment>> timeLineSegments = timeline.lookup(
+      final List<TimelineObjectHolder<String, DataSegment>> timeLineSegments = CompactionTaskUtils.toTimelineSegments(
```

Comment:
This could use `VersionedIntervalTimeline.forSegments(usedSegments)`.

Reply:
Changed.